### DISCUSSION TITLE
Polyenergetic Spectrum Capabilities

### DISCUSSION THREAD

**Discussion by Dale-Black** [JuliaImageRecon/Sinograms.jl#38] [General]
Hello. What would the process look like to bring polyenergetic spectrum capabilities from MIRT to MIRT.jl? What are some of the main steps that need to be completed before this is possible?

**Comment by JeffFessler** [comment]
The Matlab code is here:
https://github.com/JeffFessler/mirt/tree/main/ct
The first priority would be all the `xray_*` files that read the NIST tables.
You could look first to see if someone has already done that in Julia.  Maybe so:
https://github.com/usnistgov/FFAST.jl
https://pages.nist.gov/FFAST.jl/

The notation in the matlab code is probably based on this paper:
http://doi.org/10.1117/12.467189

You also need to know the spectrum of your system to make any of this useful.  If you don't know that then I'd recommend looking at these papers that work around that (for recon, not for system simulation):
- http://doi.org/10.1088/1361-6560/ac5f71
- http://doi.org/10.1109/TMI.2019.2921929
